openmc/tallies.py

```diff
@@ -259,24 +264,69 @@ def with_summary(self):
 
     @property
     def sum(self):
+        if not self._statepoint:
```

Comment:
If you were to simply return the ``sum`` here rather than set it equal to the ``_sum`` class attribute, this would automatically allow for the data to be "cleared" from the tally by the Python garbage collector if and when it is no longer needed (*e.g.*, if the statepoint file is too large to fit in memory, so a user processes tally data in "batches"). This approach would also align well with the approach you used in the statepoint and summary properties.